internal/engine/cold_start_tracker.go

```diff
@@ -0,0 +1,38 @@
+package engine
```

Comment:
all new subscribers should go in a subpackage! might make sense to put this in the same package as the other trace stuff?